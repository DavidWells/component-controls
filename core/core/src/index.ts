export * from './utils';
export { randomizeData } from './randomizeData';
export * from './smartControls';
export * from './source';
export * from './faker';
