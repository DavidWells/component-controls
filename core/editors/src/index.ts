export * from './editors/prop-factory';
export * from './editors/types';
export * from './forms/ControlsTable/ControlEditorsTable';
export { FlexContainer } from './components/FlexContainer/FlexContainer';
