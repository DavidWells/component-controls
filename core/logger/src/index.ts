import chalk from 'chalk';

export type LogLevel = 'all' | 'errors' | 'none';

export interface LogOptions {
  logLevel?: LogLevel;
  colors: [number, number, number];
}

const defaultOptions: LogOptions = {
  logLevel: 'all',
  colors: [244, 147, 66],
};

let logOptions: LogOptions = { ...defaultOptions };

const output = (heading: string, text: string) => {
  console.log(chalk.bgRgb(...logOptions.colors)(`@${heading}`), text);
};

export const log = (heading: string, text: string): void => {
  if (logOptions.logLevel === 'all') {
    output(heading, text);
  }
};
export const error = (heading: string, text: string): void => {
  if (logOptions.logLevel !== 'none') {
    output(heading, text);
  }
};

export const setLogOptions = (options: Partial<LogOptions> = {}): void => {
  logOptions = { ...logOptions, ...options };
};
