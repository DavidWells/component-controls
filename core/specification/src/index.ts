export * from './types';

export type SetControlValueFn = (
  storyId: string,
  propertyName: string | undefined,
  value: any,
) => void;
export type ClickControlFn = (storyId: string, propertyName: string) => void;
