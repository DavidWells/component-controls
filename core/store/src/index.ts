export * from './Store/Store';
export * from './Store/HMRStore';
export * from './types';
export * from './hooks';
export * from './serialization/load-store';
export * from './create-pages';
