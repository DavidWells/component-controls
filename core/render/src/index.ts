import { FrameworkRenderFn } from '@component-controls/core';
import { render as renderReact } from './react';

//noop
export const renderers: Record<string, FrameworkRenderFn> = {
  react: renderReact,
};
