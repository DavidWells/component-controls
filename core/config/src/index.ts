import * as path from 'path';
import * as fs from 'fs';
import { sync as globSync } from 'glob';
import yargs from 'yargs';
import { Configuration } from '@component-controls/specification';

export const configFileNames = ['main.js', 'main.json', 'main.ts'];

export interface ConfigrationResult {
  config: Configuration;
  configPath: string;
}
/**
 * return the configration object
 * command line accepts -c/ -config parameter for config path
 * the config file is assumed named main.js/main.ts
 * @param baseFolder project folder to start the searh with
 */
export const getConfiguration = (
  baseFolder: string,
  args: string[] = process.argv,
): ConfigrationResult | undefined => {
  const argv = yargs(args)
    .option('config', {
      alias: 'c',
      description: 'configuration folder',
      type: 'string',
    })
    .help()
    .alias('help', 'h').argv;

  const configPath = argv.config
    ? path.resolve(baseFolder, argv.config)
    : baseFolder;
  const configFiles = fs.readdirSync(configPath);
  const configFile = configFiles.find(file =>
    configFileNames.includes(file.toLowerCase()),
  );

  return configFile
    ? {
        config: require(path.resolve(configPath, configFile)),
        configPath,
      }
    : undefined;
};

/**
 * find the story files out of a configuration file
 * using glob for the regex file search
 * @param param0 configuration info return by getConfiguration
 */
export const extractStories = ({
  config,
  configPath,
}: ConfigrationResult): string[] | undefined => {
  const stories =
    config && config.stories
      ? config.stories.reduce((acc: string[], storyRg: string) => {
          const regex = storyRg.replace(/\((.*?)\)/g, '+($1)');
          return [...acc, ...globSync(path.resolve(configPath, regex))];
        }, [])
      : undefined;
  return stories;
};
