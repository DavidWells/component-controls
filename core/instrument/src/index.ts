import * as parser from '@babel/parser';
const mdx = require('@mdx-js/mdx');
import traverse from '@babel/traverse';
import generate from '@babel/generator';
import prettier, { Options, ResolveConfigOptions } from 'prettier';
import parserBabel from 'prettier/parser-babylon';
import { StoriesGroup, Story } from '@component-controls/specification';
import { extractCSFStories } from './babel/csf-stories';
import { extractMDXStories } from './babel/mdx-stories';
import { removeMDXAttributes } from './babel/remove-mdx-attributes';

type TraverseFn = (stories: StoriesGroup) => any;

export type PrettierOptions = Options & {
  resolveConfigOptions?: ResolveConfigOptions;
};
const parseSource = async (
  code: string,
  traverseFn: TraverseFn,
  originalSource: string,
  filePath?: string,
  prettierOptions?: PrettierOptions,
): Promise<StoriesGroup> => {
  let source: string;
  if (prettierOptions !== false) {
    const { resolveConfigOptions, ...otherOptions } = prettierOptions || {};
    let allPrettierOptions = otherOptions;
    if (filePath) {
      const userOptions = await prettier.resolveConfig(
        filePath,
        resolveConfigOptions,
      );
      allPrettierOptions = { ...userOptions, ...allPrettierOptions };
    }

    source = prettier.format(code, {
      parser: 'typescript',
      plugins: [parserBabel],
      ...allPrettierOptions,
    });
  } else {
    source = code;
  }

  const ast = parser.parse(source, {
    sourceType: 'module',
    plugins: ['jsx', 'typescript'],
  });
  const stories = {
    stories: {},
    source: originalSource,
  };
  traverse(ast, traverseFn(stories));
  Object.keys(stories.stories).forEach((key: string) => {
    //@ts-ignore
    const story: Story = stories.stories[key];
    const { start, end } = story.loc || {};
    if (start && end) {
      const lines = source.split('\n');

      if (start.line === end.line) {
        story.source = lines[start.line - 1].substring(
          start.column,
          end.column,
        );
      } else {
        const startLine = lines[start.line - 1];
        const endLine = lines[end.line - 1];
        if (startLine !== undefined && endLine !== undefined) {
          story.source = [
            startLine.substring(start.column),
            ...lines.slice(start.line, end.line - 1),
            endLine.substring(0, end.column),
          ].join('\n');
        }
      }
    }
  });
  return stories;
};

export const parseCSF = async (
  source: string,
  filePath?: string,
  prettierOptions?: PrettierOptions,
): Promise<StoriesGroup> => {
  return await parseSource(
    source,
    extractCSFStories,
    source,
    filePath,
    prettierOptions,
  );
};

export const parseMDX = async (
  source: string,
  filePath?: string,
  prettierOptions?: PrettierOptions,
): Promise<StoriesGroup> => {
  const code = await mdx(source);

  const ast = parser.parse(code, {
    sourceType: 'module',
    plugins: ['jsx', 'typescript'],
  });
  traverse(ast, removeMDXAttributes());
  const newCode = generate(ast, {
    retainFunctionParens: true,
    retainLines: true,
  });
  return await parseSource(
    newCode.code,
    extractMDXStories,
    source,
    filePath,
    prettierOptions,
  );
};
