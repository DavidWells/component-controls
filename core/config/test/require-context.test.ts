import { loadConfiguration, configRequireContext } from '../src/index';

describe('require-context', () => {
  it('exact names', () => {
    expect(
      configRequireContext(
        loadConfiguration(__dirname, undefined, [
          'file',
          'name',
          '--config',
          'fixtures/exact-names',
        ]),
      ),
    ).toMatchSnapshot();
  });
});
