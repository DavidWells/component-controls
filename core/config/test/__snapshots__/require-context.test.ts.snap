// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require-context exact names 1`] = `
Array [
  Object {
    "directory": "/Users/atanasster/cc/examples/stories/src/stories",
    "regExp": /controls-editors-starter\\.stories\\.tsx/,
    "useSubdirectories": false,
  },
  Object {
    "directory": "/Users/atanasster/cc/examples/stories/src/blogs",
    "regExp": /gatsby-nextjs-storybook\\.mdx/,
    "useSubdirectories": false,
  },
]
`;
